### PR TITLE
animal (sharebeef)

### PR DESCRIPTION
**Pull request by gomago1**
Initial proposal for animal data model for ShareBeef UC

### CODE REVIEW

File: specs/AgriFood/Animal/example-normalized.json [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    "breed": {
+        "value":"Merina"
+    },
+    "calvedBy": {
+        "value":"urn:ngsi-ld:Animal:aa9f1295-425c-8ba3-b745-b653097d5a87"
+    },
+    "siredBy": {
+        "value":"urn:ngsi-ld:Animal:aa9f1295-425c-8ba3-b745-b653097d5a87"
+    },
+    "location":
```
Comment (jmcanterafonseca) [inline specs/AgriFood/Animal/example-normalized.json]: be careful 

```
location: {
  "type": "geo:json",
  "value": {
    "type": "Point",
    "coordinates":
        [
            -4.754444444,
            41.640833333
        ]
   }
}
```

File: specs/AgriFood/Animal/schema.json [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+                    "description":"The owner of the animal"
+                },
+                "locatedAt":{
+                    "type":"string",
+                    "format":"uri",
+                    "description":"AgridataParcel relationship"
+                },
+                "phenologicalCondition":{
+                    "type":"string",
+                    "enum":["lactating baby","grazing baby","male adult","female adult","male young","female young"],
```
Comment (jmcanterafonseca) [inline specs/AgriFood/Animal/schema.json]: can we use just values without spaces by using camelCase, ex. 

lactatingBaby, grazingBaby, etc.

File: specs/AgriFood/Animal/schema.json [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+                    "description":"Phenological condition of the animal"
+                },
+                "reproductiveCondition":{
+                    "type":"string",
+                    "enum":["no status","inactive","inCalf","inHeat","active"],
```
Comment (jmcanterafonseca) [inline specs/AgriFood/Animal/schema.json]: check camelCase for values

File: specs/AgriFood/Animal/schema.json [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+                    "description":"Reproductive condition of the animal"
+                },
+                "healthCondition":{
+                    "type":"string",
+                    "enum":["healthy","sick","in treatment"],
```
Comment (jmcanterafonseca) [inline specs/AgriFood/Animal/schema.json]: check camel case for values

**Comment by gomago1** [comment]
I think I have corrected the errors

File: specs/AgriFood/Animal/example-normalized.json [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+        "type":"DateTime",
+        "value":"2017-05-04T12:30:00Z",
+    },
+    "sex": {
+        "value":"female"
+    },
+    "breed": {
+        "value":"Merina"
+    },
+    "calvedBy": {
```
Comment (jmcanterafonseca) [inline specs/AgriFood/Animal/example-normalized.json]: when an attribute is a Relationship, please better put it explicitly i.e. 

```
calvedBy: {
  "type": "Relationship",
  "value":"urn:ngsi-ld:Animal:aa9f1295-425c-8ba3-b745-b653097d5a87"
}
```

File: specs/AgriFood/Animal/example-normalized.json [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+        "value":"urn:ngsi-ld:Animal:aa9f1295-425c-8ba3-b745-b653097d5a87"
+    },
+    "siredBy": {
```
Comment (jmcanterafonseca) [inline specs/AgriFood/Animal/example-normalized.json]: same comment

File: specs/AgriFood/Animal/example-normalized.json [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+            [
+                -4.754444444,
+                41.640833333
+            ]
+       }
+    },
+    "weight":{
+        "value":65.3
+    },
+    "ownedBy":{
```
Comment (jmcanterafonseca) [inline specs/AgriFood/Animal/example-normalized.json]: same comment.

File: specs/AgriFood/Animal/example-normalized.json [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+        "value":"reference_to_farmer"
+    },
+    "locatedAt":"reference_to_farm/AgriParcel",
```
Comment (jmcanterafonseca) [inline specs/AgriFood/Animal/example-normalized.json]: be careful you need type and value. better if you put an explicit URN

File: specs/AgriFood/Animal/example-normalized.json [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+    "phenologicalCondition":{
+        "value":"adult"
+    },
+    "reproductiveCondition":{
+        "value":"inCalf"
+    },
+    "healthCondition":{
+        "value":"healthy"
+    },
+    "feedWith":{
```
Comment (jmcanterafonseca) [inline specs/AgriFood/Animal/example-normalized.json]: relationship

File: specs/AgriFood/Animal/example-normalized.json (hunk 7 above)
Comment (jmcanterafonseca) [reply specs/AgriFood/Animal/example-normalized.json]: explicit URN. Maybe we need here to point to  a reference of https://schema.org/Person or https://schema.org/Organization

File: specs/AgriFood/Animal/example-normalized.json (hunk 9 above)
Comment (jmcanterafonseca) [reply specs/AgriFood/Animal/example-normalized.json]: maybe we need also a Data Model for Feed

File: specs/AgriFood/Animal/example.json [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+    {
+        "type": "Point",
+        "coordinates":
+        [
+            -4.754444444,
+            41.640833333
+        ]
+    },
+    "weight":65.3,
+    "ownedBy":"reference_to_farmer",
```
Comment (jmcanterafonseca) [inline specs/AgriFood/Animal/example.json]: explicit URNs

**Comment by jmcanterafonseca** [comment]
@gomago1  another comment, please can you provide the spec in markdown format, instead of PDF? 

thanks!

**Comment by gomago1** [comment]
Hi,
I've corrected the last comments. I think we have to create the model for feed also but it is not included yet

File: specs/AgriFood/Animal/doc/animal.md [hunk 11]
```diff
@@ -0,0 +1,223 @@
...
+    - Attribute metadata:
+        - timestamp: optional timestamp for the observed value.
+            - Type: DateTime
+    - Optional
+
+### Example
+Below a JSON example of the animal data model is shown:
+```json
+{
+    "id":"8943423481234123",
```
Comment (jmcanterafonseca) [inline specs/AgriFood/Animal/doc/animal.md]: please use ```urn:ngsi-ld:Animal:8943423481234123```

File: specs/AgriFood/Animal/doc/animal.md [hunk 12]
```diff
@@ -0,0 +1,223 @@
...
+    },
+    "sex":{
+        "value":"female"
+    },
+    "breed":{
+        "value":"Meriana"
+    },
+    "calvedBy":{
+        "type":"Relationship",
+        "value":"6325415874"
```
Comment (jmcanterafonseca) [inline specs/AgriFood/Animal/doc/animal.md]: use urn:ngsi-ld

File: specs/AgriFood/Animal/doc/animal.md [hunk 13]
```diff
@@ -0,0 +1,223 @@
...
+    },
+    "siredBy":{
+        "type":"Relationship",
+        "value":"1478515874"
```
Comment (jmcanterafonseca) [inline specs/AgriFood/Animal/doc/animal.md]: use urn:ngsi-ld

File: specs/AgriFood/Animal/doc/animal.md [hunk 14]
```diff
@@ -0,0 +1,223 @@
...
+        "type":"Relationship",
+        "value":"32541789",
+        "metadata":{
+            "timestamp":{
+                "type":"DateTime",
+                "value":"2019-01-14T14:23:14.00Z"
+            }
+        }
+    },
+    "locatedAt":{
```
Comment (jmcanterafonseca) [inline specs/AgriFood/Animal/doc/animal.md]: use urn:ngsi-ld

File: specs/AgriFood/Animal/doc/animal.md [hunk 15]
```diff
@@ -0,0 +1,223 @@
...
+        "value":"145879654",
+        "metadata":{
+            "timestamp":{
+                "type":"DateTime",
+                "value":"2019-03-22T18:23:31.00Z"
+            }
+        }
+    },
+    "phenologicalCondition":{
+        "value":"lactacting animal",
```
Comment (jmcanterafonseca) [inline specs/AgriFood/Animal/doc/animal.md]: check with enumeration values

File: specs/AgriFood/Animal/doc/animal.md [hunk 16]
```diff
@@ -0,0 +1,223 @@
...
+        "metadata":{
+            "timestamp":{
+                "type":"DateTime",
+                "value":"2019-03-22T18:23:31.00Z"
+            }
+        }
+    },
+    "reproductiveCondition":{
+        "value":"nonbreeding animal",
```
Comment (jmcanterafonseca) [inline specs/AgriFood/Animal/doc/animal.md]: check with enumeration values

File: specs/AgriFood/Animal/doc/animal.md [hunk 17]
```diff
@@ -0,0 +1,223 @@
...
+        "metadata":{
+            "timestamp":{
+                "type":"DateTime",
+                "value":"2018-12-22T18:23:31.00Z"
+            }
+        }
+    },
+    "feedWith":{
+        "type":"Relationship",
+        "value":"241587",
```
Comment (jmcanterafonseca) [inline specs/AgriFood/Animal/doc/animal.md]: use urn:ngsi-ld

File: specs/AgriFood/Animal/example-normalized.json (hunk 9 above)
Comment (jmcanterafonseca) [reply specs/AgriFood/Animal/example-normalized.json]: wouldn't it be `fedWith` ?

**Comment by gomago1** [comment]
Hi,

I think I've made the changes.
Thanks

File: specs/AgriFood/Animal/example-normalized.json [hunk 18]
```diff
@@ -0,0 +1,70 @@
...
+    },
+    "reproductiveCondition":{
+        "value":"inCalf"
+    },
+    "healthCondition":{
+        "value":"healthy"
+    },
+    "fedWith":{
+        "type": "Relationship",
+        "value":"reference_to_feed"
```
Comment (jmcanterafonseca) [inline specs/AgriFood/Animal/example-normalized.json]: use URN

File: specs/AgriFood/Animal/example.json [hunk 19]
```diff
@@ -0,0 +1,29 @@
...
+    "locatedAt":"reference_to_farm/AgriParcel",
+    "phenologicalCondition":"adult",
+    "reproductiveCondition":"inCalf",
+    "healthCondition":"healthy",
+    "fedWith":"reference_to_feed",
```
Comment (jmcanterafonseca) [inline specs/AgriFood/Animal/example.json]: use URN

**Comment by gomago1** [comment]
Done

File: specs/AgriFood/Animal/schema.json [hunk 20]
```diff
@@ -0,0 +1,99 @@
...
+                "welfareCondition":{
+                    "type":"string",
+                    "enum":["issue","adequate"],
+                    "description":"Indicator of the animal welfare"
+                }
+            }
+        }
+    ],
+    "required": [
+        "id",
```
Comment (jmcanterafonseca) [inline specs/AgriFood/Animal/schema.json]: "type" should also be required

**Comment by jmcanterafonseca** [comment]
@gomago1 please also check your JSON schema and JSON files, CI is still reporting errors

**Comment by gomago1** [comment]
Changes done

**Comment by jmcanterafonseca** [comment]
@gomago1 still CI failing, please check that your examples are accepted by Orion

**Comment by gomago1** [comment]
Hi,

There is something that I can't understand. Checking the CI, I have found the following errors:

image

If I explore them, some errors appear but I do not know how to link them with the documents that I created. I'm afraid that I need a little bit help.

Thanks

File: specs/AgriFood/Animal/example-normalized.json [hunk 21]
```diff
@@ -0,0 +1,70 @@
...
+    "weight":{
+        "value":65.3
+    },
+    "ownedBy":{
+        "type": "Relationship",
+        "value":"http://person.org/leon"
+    },
+    "locatedAt":{
+        "type": "Relationship",
+        "object": "urn:ngsi-ld:AgriParcel:1ea0f120-4474-11e8-9919-672036642081"
```
Comment (jmcanterafonseca) [inline specs/AgriFood/Animal/example-normalized.json]: in NGSIv2 this should be `value` instead of`object`. In NGSI-LD yes, it is `object`

**Comment by gomago1** [comment]
Now, there is just one error in the Documentation Tests but I do not know how to solve it

**Comment by jmcanterafonseca** [comment]
@gomago1 you have warnings over your markdown content. Please can you try running over your markdown code? Thanks! 

prettier spec.md --tab-width 4 --print-width 80 --write --prose-wrap always

**Comment by gomago1** [comment]
Hi Jose Manuel,

I have tried it with Visual Studio and Atom and it does not work, even though I see it well. It should be a format error about indent
image

I attach the file

[animal.zip](https://github.com/FIWARE/dataModels/files/3127168/animal.zip)

**Comment by jmcanterafonseca** [comment]
we are going to merge and then fix the markdown problem

**Pull request merged**